converse.js

```diff
@@ -2617,6 +2617,11 @@
                 var $feedback = this.$('.conn-feedback'); // we want to still show any existing feedback.
                 this.$el.html(converse.templates.controlbox(this.model.toJSON()));
                 var cfg = {'$parent': this.$el.find('.controlbox-panes'), 'model': this};
+
+                if (!this.model.get('connected') && converse.auto_login){
+                    converse.auto_login = false;
+                }
+
```

Comment:
Thanks, but I'm not sure I like this check here. I think a better place would be the [logOut](https://github.com/jcbrand/converse.js/blob/875a05e0a5cceafce7befdaa64eacc357efa44dc/converse.js#L756) method. Then it's more clear that the `auto_login` gets set to `false` because the user logged out.